client/app/bundles/course/achievement/badge-uploader.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { PropTypes } from 'react';
+import { injectIntl, defineMessages, intlShape } from 'react-intl';
+import Avatar from 'material-ui/Avatar';
+import Badge from 'material-ui/Badge';
+import Dropzone from 'react-dropzone';
+import IconButton from 'material-ui/IconButton';
+import CloseIcon from 'material-ui/svg-icons/navigation/close';
+
+const styles = {
+  badge: {
+    height: '120px',
+    marginRight: '1em',
+    width: '120px',
+  },
+  badgeIconActive: {
+    backgroundColor: 'rgb(200,200,200)',
+  },
+  badgeIconInactive: {
+    backgroundColor: 'rgb(225,225,225)',
+  },
+  dropzone: {
+    backgroundColor: 'rgb(225,225,225)',
+    borderRadius: '5px',
+    height: '200px',
+    paddingTop: '30px',
+    textAlign: 'center',
+    width: '100%',
+  },
+};
+
+const translations = defineMessages({
+  badgeUploaderTitle: {
+    id: 'course.achievement.badgeUploaderTitle',
+    defaultMessage: 'Click to select a new file, or drag and drop it into the zone.',
+  },
+});
+
+const propTypes = {
```

Comment:
Put this as a static proptype of BadgeUploader instead - this way, it is kept at the top of the component definition, and we don't need to have the `BadgeUploader.propTypes = propTypes;`.